scripts/codehash.js

```diff
@@ -0,0 +1,6 @@
+const { exec } = require('child_process');
+
+const defaultAddress = process.env.DEFAULTADDRESS;
+const rpc = process.env.RPCURL;
+
+exec('echo $(cast keccak $(cast code ' + defaultAddress + ' --rpc-url ' + rpc + ') > codehash.txt)');
```

Comment:
I don't understand why this is an NPM script.

Also, if we use NPM scripts, lets use the `ethers` dependency that we already include instead of shelling out to an external tool.

Reply:
Mentioned in the PR Description:
> adds a JS script for checking the code hash of the contract as GitHub Actions have a limit on the length of the arguments.

Reply:
> adds a JS script for checking the code hash of the contract as GitHub Actions have a limit on the length of the arguments.

This doesn't make sense to me :see_no_evil: - what limit were we hitting exactly?

My point still stands though - I wouldn't use cast and use Ethers.js directly here. Shelling is kind of unsafe especially since `rpc` is a parameter from an external untrusted source.

Reply:
Oh, I see `cast keccak $(cast code ...)` is being replaced with `cast keccak 0x.....` for a very long `0x....`.

You can just solve this with piping though: `cast code ... | cast keccak` (at least, as long as `cast keccak` takes in hex from stdin).

Reply:
Looks like cast doesn't like the trailing newline in the code hex output. This can be fairly easily solved with `tr`:

```
cast code 0x41675C099F32341bf84BFc5382aF534df5C7461a --rpc-url https://eth.llamarpc.com | tr -d '\n' | cast keccak
```

Reply:
Awesome! Updated!